src/core/ledger/grainIntegration.js

```diff
@@ -0,0 +1,124 @@
+// @flow
+
+import {Ledger, type PayoutAddress, type CurrencyId} from "./ledger";
+import {type Distribution} from "./distribution";
+import {getDistributionBalances} from "./distributionSummary/distributionSummary.js";
+import {type Currency} from "./currency.js";
+import {type IdentityId} from "../identity";
+import * as G from "./grain";
+
+type Transfer = {|
+  amount: G.Grain,
+  memo: string,
+|};
+
+export type AccountDistributions = Map<PayoutAddress, G.Grain>;
+export type PayoutAddressToId = Map<PayoutAddress, IdentityId>;
+export type TransferredGrain = Map<PayoutAddress, Transfer>;
+
+export type IntegrationResult = {|
```

Comment:
Integration still feels too vague to me. "Payout" is a lot more descriptive, and could probably be substituted in most places.

Reply:
We'll have integrations that don't actually pay out though, for like logging or record keeping purposes